pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go

```diff
@@ -66,32 +67,62 @@ func (c *cloudProviderObserver) ObserveCloudProviderNames(genericListers configo
 		return existingConfig, append(errs, err)
 	}
 
+	// Set cloudProvider value
 	cloudProvider := getPlatformName(infrastructure.Status.Platform, recorder)
 	if len(cloudProvider) > 0 {
 		if err := unstructured.SetNestedStringSlice(observedConfig, []string{cloudProvider}, c.cloudProviderNamePath...); err != nil {
 			errs = append(errs, err)
 		}
 	}
 
+	// Set cloudProviderConfig value
+	existingCloudConfig, _, err := unstructured.NestedStringSlice(existingConfig, c.cloudProviderConfigPath...)
+	if err != nil {
+		errs = append(errs, err)
+		// keep going on read error from existing config
+	}
+
+	cloudProviderConfig, configErrs := c.getCloudProviderConfig(genericListers, recorder, infrastructure)
+	if configErrs != nil {
+		errs = append(errs, configErrs...)
+	}
```

Comment:
How is the cloud provider config handled currently? Surely this is handled in MCO somehow already? How does this tie in when the cloud provider is external?

Reply:
[Here](https://github.com/openshift/cluster-kube-controller-manager-operator/blob/35aa6d524d9707699630f04565d7250ccf9118a0/pkg/operator/configobservation/configobservercontroller/observe_config_controller.go#L82) is how it is set in KCMO. This method will co-exist on the same list of observers, and will be placed on the next line. Those two methods logic is exclusive, either `NewCloudVolumePluginObserver` sets the cloud-provider flag (well `--external-cloud-volume-plugin`) or `NewCloudProviderObserver` does it.  MCO handles it differently, we don't use it there.